forecasting/src/autogluon/forecasting/trainer/auto_trainer.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from typing import Dict, Union, Optional, Any
+
+from ..models.presets import get_preset_models
+from .abstract_trainer import AbstractForecastingTrainer, TimeSeriesDataFrame
+
+logger = logging.getLogger(__name__)
+
+
+class AutoForecastingTrainer(AbstractForecastingTrainer):
+    def construct_model_templates(self, hyperparameters, **kwargs):
+        path = kwargs.pop("path", self.path)
+        eval_metric = kwargs.pop("eval_metric", self.eval_metric)
+        quantile_levels = kwargs.pop("quantile_levels", self.quantile_levels)
+        hyperparameter_tune = kwargs.get("hyperparameter_tune", False)
+        return get_preset_models(
+            path=path,
+            eval_metric=eval_metric,
+            prediction_length=self.prediction_length,
+            freq=self.freq,
+            hyperparameters=hyperparameters,
+            hyperparameter_tune=hyperparameter_tune,
+            quantiles=quantile_levels,
+            invalid_model_names=self._get_banned_model_names(),
+        )
+
+    # todo: implement cross-validation / holdout strategy
+    # todo: including CVSplitter logic
+    def fit(
+        self,
+        train_data: TimeSeriesDataFrame,
+        hyperparameters: Union[str, Dict[Any, Dict]],
+        val_data: Optional[TimeSeriesDataFrame] = None,
+        hyperparameter_tune: bool = False,
+        time_limit: float = None,
+        infer_limit: float = None,  # todo: implement
```

Comment:
nit: ditto- `train_data` and `val_data`, consider standardizing parameters order.